### ISSUE TITLE
Externalize the HDF5 magic of the setup.py in a submodule

### ISSUE THREAD

**Issue by wizmer** [h5py/h5py#1486]
Hi,

I am the maintainer of https://github.com/BlueBrain/MorphIO/ 
It's a neuroscience IO library and one of the supported input is H5 files.
We are regularly facing some issues related to delivering libHDF5.so, in particular when providing python wheels.
Since I will have to work on these issues anyway, and that I will probably copy paste a lot of h5py code, I was thinking about putting all the shared library handling magic in a different repository that both h5py and morphio would use as a submodule.
Do you think that would be feasible/a good idea ? I could dedicate some of my working time to it.

Thanks for your input

**Comment by tacaswell** [comment]
Can you just depend on h5py?  If you are building wheels with libhdf5 bundled in can lead to chaos if your version and h5py's version are out of sync (we have historically had this problem between h5py and pytables).

I am :-1: on making our build chain part of the public API.  If it is public, then we can not change it at will.

Much of the wheel building and packing logic is actually in https://github.com/matthew-brett/multibuild, that may be what you are looking for?

**Comment by wizmer** [comment]
I understand your point of view. Yes, I'm aware of multibuild. I actually asked for a bump of the default HDF5 version [there](https://github.com/matthew-brett/multibuild/pull/263). But my issues are mostly related to the multibuild generated wheels.

I'll try to incorporate h5py in my build to get libHDF5 from there as you suggested. I'll let you know if anything useful comes out.

**Comment by takluyver** [comment]
> I am :-1: on making our build chain part of the public API. 

:-1: from me too. We've been changing it quite a bit recently, and I don't think it's doing anything particularly beautiful. If you want to reuse the build scripts specifically, I think copy and modify is the best way to do so.

**Comment by takluyver** [comment]
Closing as we seem to agree this is not the way h5py wants to go. Feel free to continue discussion here, we can always reopen it if we do want to do something.

**Comment by wizmer** [comment]
Hi, it's me again. Sorry to bother. I have a quick question that does not deserve opening a new ticket. I wanted to get some inspiration from the way h5py build the Windows wheels but I haven't been able to find which stack build them. I had a look a the appveyor file but I saw the commit and issue that comments the code building the wheels there (saying those wheels were just there for debugging purposes). I also had a look at the h5py-wheel repo but there is no appveyor file there. Could you indicate me how are built the Windows wheels that end-up on PyPi ? Thanks a lot.

**Comment by takluyver** [comment]
I think the Windows wheels are built by Christoph Gohlke, though I'd like to automate the process more. See also #1300.

**Comment by tacaswell** [comment]
Correct, Christoph Gohlke builds the windows wheels and then we upload them to pypi.
